content/gsoc/2022.md

```diff
@@ -0,0 +1,600 @@
+- Final term: Implement the way to show the API when hovered over some interface control, create documentation.
+
+### Mentors
+- thestr4ng3r
+- Megabeets
+
+### Links/Resources
+- [SDB Module/API for Cutter Python/Jupyter integration](https://github.com/rizinorg/cutter/issues/1778)
+- [Jupyter plugin for Cutter](https://github.com/rizinorg/cutter-jupyter)
+
+## Multi-Tasking and Event-driven architecture (Extended deadline)
+
+Cutter is a reverse engineering framework that is powered by Rizin. The information it gets about functions, strings, imports, and the analysis are all performed in Rizin and displayed in Cutter. Currently, Cutter is pulling information from Rizin only on demand. This is problematic because sometimes the user performs changes (via plugins, the console widget, and more) that are affecting the information from Rizin, but Cutter doesn't know about these changes to apply the to the UI. For example, if  a user will define a new function in a Python script or via the console widget by using the Rizin command `af @ <addr>`, Cutter will now show this new function in the Functions widget until the user will refresh the interface manually (edit -> Refresh Contents).
+
+In addition, this task will also handle the analysis in the background feature, to allow the analysis performed by Rizin to happen while the interface is active.
+
+### Tasks
+The overall implementation of this task should start from Rizin by adding events to many of the functions. This can be done using `rz_events`. For example, add an even for function creating, for section creation, for flag deletion, for name changed, and more
+
+- Add events to all the relevant functions inside Rizin
+- Add support for these events in Cutter and refresh and update the relevant widgets per each event
+- Support analysis in the background and allow the user to start its session while Rizin is analyzing (see [#1856](https://github.com/rizinorg/cutter/issues/1856), [#1574](https://github.com/rizinorg/cutter/issues/1574))
+
+### Skills
+The participant should be comfortable with the C++ for Cutter and C for Rizin. The participant should be familiar with Qt framework.
+
+### Difficulty
+Advanced
+
+### Benefits for the participant
+The participant will gain an experience of creating complex event-driven software in both C and C++ languages.
+
+### Benefits for the project
+It will allow to work on big files effortlessly in Cutter, will improve analysis quality as well.
+
+### Assess requirements for midterm/final evaluation
+- 1st term: Implement events everywhere in the relevant places across Rizin code and event-driven
+	interaction with Cutter.
+- Final term: Add support for the Cutter interface refresh based on the events from Rizin, implement analysis in background.
+
+### Mentors
+- thestr4ng3r
+- Karliss
+
+## Heap viewer (Medium project)
```

Comment:
Wasn't that partially finished last year?

Reply:
There are major performance issues with the windows implementation and [Cutter integration wasn't completed](https://github.com/rizinorg/cutter/pull/2723). This could only be a medium project if support for other allocators is included.